File: world-beauty/src/WB-ATV1/src/Models/CPF.ts
export default class CPF {
    private valor: string;
    private dataEmissao: Date;

    constructor(valor: string, dataEmissao: Date) {
        this.valor = valor.replace(".", "").replace("-", "");
        this.dataEmissao = dataEmissao;
    }

    public get GetValor(): string {
        return this.valor;
    }

    public get GetValorFormatado(): string {
        // XXX.XXX.XXX-XX
        return `${this.valor.slice(0, 3)}.${this.valor.slice(3, 6)}.${this.valor.slice(6, 9)}-${this.valor.slice(9, 12)}`;
    }

    public get GetDataEmissao(): Date {
        return this.dataEmissao;
    }
}

File: backend/WorldBeauty/CPF.java
package WorldBeauty;

import java.util.Date;

public class CPF {
    private String valor;
    private Date dataEmissao;

    CPF(String valor, Date dataEmissao) {
        this.valor = valor.replace(".", "").replace("-", "");
        this.dataEmissao = dataEmissao;
    }

    public String GetValor() {
        return this.valor;
    }

    public String GetValorFormatado() {
        // XXX.XXX.XXX-XX
        return this.valor.substring(0, 3) + "." + this.valor.substring(3, 6) + "." + this.valor.substring(6, 9) + "-" + this.valor.substring(9, 12);
    }

    public Date GetDataEmissao() {
        return this.dataEmissao;
    }
}
